From cb98ed45f51225b2791bdfaba388391d91265434 Mon Sep 17 00:00:00 2001
From: Mathieu Malaterre <mathieu.malaterre@gmail.com>
Date: Wed, 30 Mar 2022 16:04:14 +0200
Subject: [PATCH] Revert "Remove some old unused and confusing code"

This reverts commit 222f20d8bcdbee861f3142d842148e74ccfb4991.

This code is still in use on BigEndian arch. For example

```
138         return Tag( Swap(val.GetGroup()), Swap(val.GetElement()) );
(gdb)
500         T swappedData = SwapperNoOp::Swap(data[0]);
```
---
 Source/Common/gdcmSwapper.txx | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/Source/Common/gdcmSwapper.txx b/Source/Common/gdcmSwapper.txx
index fe32d6c27..d63ccd5d7 100644
--- a/Source/Common/gdcmSwapper.txx
+++ b/Source/Common/gdcmSwapper.txx
@@ -133,7 +133,10 @@ namespace gdcm
     return tempF;
     }
 
-  template <> inline Tag SwapperNoOp::Swap<Tag>(Tag val);
+  template <> inline Tag SwapperNoOp::Swap<Tag>(Tag val)
+    {
+    return Tag( Swap(val.GetGroup()), Swap(val.GetElement()) );
+    }
 
   template <> inline void SwapperNoOp::SwapArray(uint8_t *, size_t ) {}
 
@@ -206,7 +209,10 @@ namespace gdcm
     return tempF;
     }
 
-  template <> inline Tag SwapperDoOp::Swap<Tag>(Tag val);
+  template <> inline Tag SwapperDoOp::Swap<Tag>(Tag val)
+    {
+    return Tag( Swap((uint32_t)val.GetElementTag()) );
+    }
 
   template <> inline void SwapperDoOp::SwapArray(uint8_t *, size_t ) {}
 
